cirq-google/cirq_google/workflow/timing.py

```diff
@@ -0,0 +1,46 @@
+# Copyright 2022 The Cirq Developers
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Utilities for timing execution."""
```

Comment:
I'm not a huge fan of this module coupling it's functionality to RunTimeInfo just so we can mutate a key in some dictionary for a specific type of object. It feels like a weird hybrid of accessor and mutator where we could very easily just have one or the other. 

If all we need to do is get timing information for RuntTimeInfo why not just add something like:

```python3
@contextlib.contextmanager
def _add_time_to_runtime(cur_runtime) -> None:
    start = time.time()
    yield
    cur_runtime.whatever = time.time() - start
```
to the top of the `quantum_runtime.py` module and use that instead ? Or make a more general timer that doesn't couple to this part of the codebase ?

Reply:
changed to a private context manager. changed to use decorator instead of class.